docs/guides/integration/jupyter.md

```diff
@@ -0,0 +1,146 @@
+# Using uv with Jupyter
+
+The [Jupyter](https://jupyter.org/) notebook is a popular tool for interactive computing, data
+analysis, and visualization. You can use Jupyter with uv in a few different ways, either to interact
+with a project, or as a standalone tool.
+
+## Using Jupyter within a project
+
+If you're working within a [project](../../concepts/projects.md), you can kick off a Jupyter server
+with access to the project's virtual environment via the following:
+
+```console
+$ uv run --with jupyter jupyter lab
```

Comment:
Future: I wonder if we want `uvx --with-project jupyter lab`?

Reply:
Yeah I kinda like that. The repeated `jupyter` would be nice to avoid.